datafusion/core/src/execution/context.rs

```diff
@@ -1035,6 +1037,65 @@ impl FunctionRegistry for SessionContext {
     }
 }
 
+#[async_trait]
+pub trait Reader<'a, T>: Sized {
+    async fn read_csv(&self, table_paths: T, options: CsvReadOptions<'_>) -> Result<DataFrame>
+    where 'a:'async_trait
+    ;
+}
+
+#[async_trait]
+impl<'a> Reader<'a, &'a str> for SessionContext {
+    async fn read_csv(&self, table_path: &'a str, options: CsvReadOptions<'_>) -> Result<DataFrame> {
+        self.read_csv(vec![table_path], options).await
+    }
+}
+
+#[async_trait]
+impl<'a> Reader<'a, String> for SessionContext {
```

Comment:
We can't use AsRef<str> here. We have to implement both `&str` and `String` seperately.